support-lib/wasm/Future_wasm.hpp

```diff
@@ -105,4 +105,12 @@ class FutureAdaptor
     }
 };
 
+template<typename U>
+struct ExceptionHandlingTraits<FutureAdaptor<U>> {
+    static em::val handleNativeException(const std::exception& e) {
+        auto r = FutureAdaptor<U>::NativePromiseType::reject(std::current_exception());
```

Comment:
How come we use current_exception here instead of e? Should we do that in the other specializations of ExceptionHandlingTraits too?

Reply:
Because when we call this function:
```
    template <typename E>
    void setException(const E& ex) {
        setException(std::make_exception_ptr(ex));
    }
```

E will be deduced to `std::exception` .

`make_exception_ptr` has this signature:

```
template< class E >
std::exception_ptr make_exception_ptr( E e ) noexcept;
```
And it will slice our object into a `std::exception`, losing all the context data we put in it.

Reply:
Basically `setException` should not be called with a polymorphic reference.  It needs to be the actual type to avoid exception being sliced.

Reply:
Sounds good, I guess just keeping it consistent helps but I understand the constraint now